internal/venice-common/src/main/java/com/linkedin/venice/pubsub/adapter/kafka/producer/ApacheKafkaProduceResultFuture.java

```diff
@@ -0,0 +1,44 @@
+package com.linkedin.venice.pubsub.adapter.kafka.producer;
+
+import com.linkedin.venice.pubsub.api.ProduceResult;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+
+public class ApacheKafkaProduceResultFuture implements Future<ProduceResult> {
```

Comment:
Same as the above: we need to transform inputs and outputs to and from a specific pubsub client format so that Venice writer can write and read without any changes to it if we switch backend pubsub system